www/js/survey/enketo/service.js

```diff
@@ -119,16 +110,8 @@ angular.module('emission.survey.enketo.service', [
    * @returns {string} answer string
    */
   function _restoreAnswer() {
-    if (_state.opts.trip) {
-      const answer = _state.opts.trip.userInput["SURVEY"];
-      return answer ? answer.data.xmlResponse : null;
-    } else if (_state.opts.prev_demographic_survey) {
-        const answer = _state.opts.prev_demographic_survey;
-        // TODO: Figure out how to retrieve and match the profile survey
-        return answer? answer.data.xmlResponse : null;
-    } else {
-        return null;
-    }
+    const answer = _state.opts.prefilledSurveyResponse || _state.opts.trip?.userInput["SURVEY"];
```

Comment:
`prefilledSurveyResponse` is only supplied for enketo-demographics and add-note-button
`userInput['SURVEY']` is only used for trip confirm / multilabel

We don't have a use case where both are defined. However in the case we did, I think it makes sense to prefer the prefilledSurveyResponse because it's been specifically defined

Reply:
Now that we want to support edits, I believe we will in fact have a use case in which both are defined. And in general, we should prioritize the prior user input over the partially filled template